File: Java/src/Cracking The Coding Interview/One_Three.java
package Ch1_InterviewQuestions;

import java.util.Scanner;

/*
 * 1.3 Given two strings, write a method to decide if one is a permutation of the other.
 * Did so using a small hash table
 */

public class One_Three 
{

	
	
	public static void main(String[] args) 
	{
		Scanner in = new Scanner(System.in);
		String perm1 = in.nextLine();
		String perm2 = in.nextLine();
		System.out.println(isPermutation(perm1, perm2));
	}
	
	static boolean isPermutation(String a, String b)
	{
		String[] alphabet = new String[1000];
		initString(alphabet); //is this needed?
		int hashNum = 0;
		for(int i = 0; i < a.length(); i++)
		{
			hashNum = (int)a.charAt(i);
			alphabet[hashNum] += a.charAt(i);
		}
		for(int i = 0; i < b.length(); i++)
		{
			hashNum = (int)b.charAt(i);
			alphabet[hashNum] += b.charAt(i);
		}
		for(int i = 0; i < 100; i++)
		{
			if(alphabet[i].length() % 2 != 0)
			{
				return false;
			}
		}
		return true;
	}

	static void initString(String[] a)
	{
		for(int i = 0; i < 1000; i++)
		{
			a[i] = "";
		}
	}
}

File: Python/src/Cracking the Coding Interview Problems/One_Three.py
"""1.3 Given two strings, write a method to decide if one is a permutation of the other."""


def main():
    s1 = raw_input("Please Enter A String: ")
    s2 = raw_input("Enter another to see if it is permutation of the first string: ")
    print is_permutation(s1, s2)
    return


def is_permutation(s1, s2):
    # do this by hash tables? AW YEAH BABY
    hash_table1 = [0 for j in range(500)]
    hash_table2 = [0 for j in range(500)]
    for a in range(0, s1.__len__()):
        hash_table1[ord(s1[a])] += 1
    for b in range(0, s2.__len__()):
        hash_table2[ord(s2[b])] += 1
    for c in range(0, 500):
        if hash_table1[c] != hash_table2[c]:
            return False
    return True

main()
